iquip/apps/builder.py

```diff
@@ -1,32 +1,212 @@
 """App module for editting the build arguments and submitting the experiment."""
 
 import json
-from typing import Any, Callable, Dict, Optional, Tuple
+from typing import Any, Callable, Dict, Optional, Tuple, Union
 
 import requests
 from PyQt5.QtCore import QObject, Qt, QThread, pyqtSignal, pyqtSlot
 from PyQt5.QtWidgets import (
-    QPushButton, QVBoxLayout, QWidget
+    QCheckBox, QComboBox, QDoubleSpinBox, QHBoxLayout, QLabel, QLineEdit, QListWidget,
+    QListWidgetItem, QPushButton, QVBoxLayout, QWidget
 )
 
 import qiwis
 from iquip.protocols import ExperimentInfo
 
+class _BaseEntry(QWidget):
+    """Base class for all argument entries.
+
+    This is a wrapper of each entry.
+    In each subclass, value() must be implemented to return the selected value.
+
+    Attributes:
+        name: The argument name.
+        argInfo: The dictionary with the argument options.
+        nameLabel: The label for showing the argument name.
+    """
+
+    def __init__(self, name: str, argInfo: Dict[str, Any], parent: Optional[QWidget] = None):
+        """Extended.
+        
+        Args:
+            name, argInfo: See the attributes section in _BaseEntry.
+        """
+        super().__init__(parent=parent)
+        self.name = name
+        self.argInfo = argInfo
+        # widgets
+        self.nameLabel = QLabel(name, self)
+        # layout
+        self.layout = QHBoxLayout(self)
+        self.layout.addWidget(self.nameLabel)
+        self.setLayout(self.layout)
```

Comment:
Usually you did not use `setLayout()`, did you? Do you know what the difference is?

Reply:
I tried to remove it, but I forgot😅
As I understand, `setLayout()` is unnecessary because it is already done while `layout = QHBoxLayout(self)`